app/main/views.py

```diff
@@ -0,0 +1,156 @@
+# coding=utf-8
+import json
+
+from flask import (render_template, request, redirect,
+                   url_for, current_app, session)
+from flask_login import login_required, logout_user, login_user, UserMixin
+
+from . import main
+from .forms import RegistrationForm, LoginForm
+from app import login_manager
+
+
+class User(UserMixin):
+    def __init__(self, user_id):
+        self.id = user_id
+
+
+def update_context(data, context):
+    """
+    this function updates context dictionary
+    with data passed in as argument.
+    :param data: 
+    :param context: 
+    :return dict: 
+    """
+    for key, value in data.items():
+        context[key] = value
+    return context
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    try:
+        return User(session[user_id])
+    except KeyError:
+        return None
+
+
+@main.route('/', strict_slashes=False)
+@main.route('/index/', strict_slashes=False)
+def index():
+    if "user_id" in session:
+        return redirect("/main/dashboard")
+    return render_template("index/index.html")
+
+
+@main.route("/homepage", strict_slashes=False)
+@main.route('/register/', strict_slashes=False)
+@main.route('/login/', strict_slashes=False)
+def homepage():
+    # renders the index page when any of the above route is reached
+    if "user_id" in session:
+        return redirect("/main/dashboard")
+    register_form = RegistrationForm()
+    login_form = LoginForm()
+    context = {"title": "Nitly", "registerForm": register_form,
+               "loginForm": login_form}
+    return render_template("index/homepage.html", context=context)
+
+
+@main.route('/start_session/', methods=["POST"], strict_slashes=False)
+def login():
+    data = request.json
```

Comment:
the  view doesn't receive a form, rather it receives a post request from an ajax call.